instrumentation/powerjob-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/powerjob/v4_0/PowerJobBasicProcessorTest.java

```diff
@@ -0,0 +1,366 @@
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.MAP_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.MAP_REDUCE_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.PYTHON_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.SHELL_PROCESSOR;
+import static io.opentelemetry.javaagent.instrumentation.powerjob.v4_0.PowerJobConstants.SPRING_DATASOURCE_SQL_PROCESSOR;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static java.util.Arrays.asList;
+
+import com.alibaba.fastjson.JSONArray;
+import com.alibaba.fastjson.JSONObject;
+import com.zaxxer.hikari.HikariDataSource;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.internal.StringUtils;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.testing.assertj.AttributeAssertion;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import tech.powerjob.official.processors.impl.FileCleanupProcessor;
+import tech.powerjob.official.processors.impl.HttpProcessor;
+import tech.powerjob.official.processors.impl.script.PythonProcessor;
+import tech.powerjob.official.processors.impl.script.ShellProcessor;
+import tech.powerjob.official.processors.impl.sql.DynamicDatasourceSqlProcessor;
+import tech.powerjob.official.processors.impl.sql.SpringDatasourceSqlProcessor;
+import tech.powerjob.worker.core.processor.TaskContext;
+import tech.powerjob.worker.core.processor.WorkflowContext;
+import tech.powerjob.worker.core.processor.sdk.BasicProcessor;
+import tech.powerjob.worker.log.OmsLogger;
+
+class PowerJobBasicProcessorTest {
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Test
+  void testBasicProcessor() throws Exception {
+    long jobId = 1;
+    String jobParam = "abc";
+    TaskContext taskContext = genTaskContext(jobId, jobParam);
+    BasicProcessor testBasicProcessor = new TestBasicProcessor();
+    testBasicProcessor.process(taskContext);
+    testing.waitAndAssertTraces(
+        trace -> {
+          trace.hasSpansSatisfyingExactly(
+              span -> {
+                span.hasName(String.format("%s.process", TestBasicProcessor.class.getSimpleName()));
+                span.hasKind(SpanKind.INTERNAL);
+                span.hasStatus(StatusData.unset());
+                span.hasAttributesSatisfying(
```

Comment:
usually we use `hasAttributesSatisfyingExactly`

Reply:
Done